### PR TITLE
fix: make 'ModelInput' field optional for inference recommendation

### PR DESCRIPTION
**Pull request by rahven14**
*Issue #, if available:* N/A

*Description of changes:* Make 'ModelInput' field optional for inference recommendation and refactor code for better readability.

*Testing done:* units

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/utils.py [hunk 1]
```diff
@@ -749,32 +749,57 @@ def update_container_with_inference_params(
...
+            }
+        )
+    else:
+        obj.update(
+            {
+                "Framework": framework,
+                "FrameworkVersion": framework_version,
+                "NearestModelName": nearest_model_name,
+            }
+        )
```
Comment (saumitravikram) [inline src/sagemaker/utils.py:805]: nit : Can the obj be partially updated also? Then we can refactor the code saying update the 3 fields(Framework, FrameworkVersion,  NearestModelName) first and then if `data_input_configuration` is not none then update `ModelInput`

File: tests/unit/test_estimator.py [hunk 2]
```diff
@@ -3319,7 +3317,6 @@ def test_register_inference_image(sagemaker_session):
     framework = "TENSORFLOW"
     framework_version = "2.9"
     nearest_model_name = "resnet50"
-    data_input_config = '{"input_1":[1,224,224,3]}'
```
Comment (saumitravikram) [inline tests/unit/test_estimator.py:3322]: Have atleast 1 new test where we check for data_input_config if not already present.

Comment (rahven14) [reply tests/unit/test_estimator.py:3322]: It is already present in the existing test files.

File: src/sagemaker/utils.py (hunk 1 above)
Comment (rahven14) [reply src/sagemaker/utils.py:805]: Yes, this can be refactored. Thank you for pointing this out!

**Review by saumitravikram — APPROVED** [review]

**Review by mufaddal-rohawala — APPROVED** [review]

**Pull request merged**